lms/djangoapps/commerce/api.py

```diff
@@ -52,25 +72,43 @@ def create_order(self, user, sku):
 
         Returns a tuple with the order number, order status, API response data.
         """
-        headers = {
-            'Content-Type': 'application/json',
-            'Authorization': 'JWT {}'.format(self._get_jwt(user))
-        }
+        def create():
+            """Internal service call to create an order. """
+            headers = {
+                'Content-Type': 'application/json',
+                'Authorization': 'JWT {}'.format(self._get_jwt(user))
+            }
+            url = '{}/orders/'.format(self.url)
+            return requests.post(url, data=json.dumps({'sku': sku}), headers=headers, timeout=self.timeout)
+        return self._call_ecommerce_service(create)
 
-        url = '{}/orders/'.format(self.url)
+    @staticmethod
+    def _call_ecommerce_service(call):
+        """
+        Makes a call to the E-Commerce Service. There are a number of common errors that could occur across any
+        request to the E-Commerce Service that this helper method can wrap each call with. This method helps ensure
+        calls to the E-Commerce Service will conform to the same output.
 
+        Arguments
+            call    --  A callable function that makes a request to the E-Commerce Service.
+
+        Returns a tuple with the order number, order status, API response data.
+        """
         try:
-            response = requests.post(url, data=json.dumps({'sku': sku}), headers=headers, timeout=self.timeout)
+            response = call()
             data = response.json()
         except Timeout:
             msg = 'E-Commerce API request timed out.'
             log.error(msg)
             raise TimeoutError(msg)
-
         except ValueError:
             msg = 'E-Commerce API response is not valid JSON.'
             log.exception(msg)
             raise InvalidResponseError(msg)
+        except Exception:
+            msg = 'E-Commerce API response resulted in an exception.'
+            log.exception(msg)
+            raise InvalidResponseError(msg)
```

Comment:
I would reconsider this as I don't think it adds much value and adds developer-facing hassle because the original exception will be lost in a log file.  If we just let unanticipated exceptions fall through we avoid making assumptions about their nature/cause and let the api caller decide what do to, which seems more usefully transparent.